core_impl/src/main/java/io/opencensus/stats/MutableAggregation.java

```diff
@@ -0,0 +1,348 @@
+     * The minimum of the population values.
+     *
+     * @return The minimum of the population values.
+     */
+    double getMin() {
+      return min;
+    }
+
+    /**
+     * The maximum of the population values.
+     *
+     * @return The maximum of the population values.
+     */
+    double getMax() {
+      return max;
+    }
+
+    /**
+     * Put a new value into the Range. Sets min and max values if appropriate.
+     *
+     * @param value the new value
+     */
+    @Override
+    void add(double value) {
+      if (value < min) {
+        min = value;
+      }
+      if (value > max) {
+        max = value;
+      }
+    }
+
+    @Override
+    final <T> T match(
+        Function<? super MutableSum, T> p0,
+        Function<? super MutableCount, T> p1,
+        Function<? super MutableHistogram, T> p2,
+        Function<? super MutableRange, T> p3,
+        Function<? super MutableMean, T> p4,
+        Function<? super MutableStdDev, T> p5,
+        Function<? super MutableAggregation, T> defaultFunction) {
+      return p3.apply(this);
+    }
+  }
+
+  /** Calculate mean on aggregated {@code MeasureValue}s. */
+  static final class MutableMean extends MutableAggregation {
+
+    private double mean = 0.0;
+    private long count = 0;
+
+    private MutableMean() {
+    }
+
+    /**
+     * Construct a {@code MutableMean}.
+     *
+     * @return an empty {@code MutableMean}.
+     */
+    static MutableMean create() {
+      return new MutableMean();
+    }
+
+    @Override
+    void add(double value) {
+      count++;
+      double deltaFromMean = value - mean;
+      mean += deltaFromMean / count;
+    }
+
+    /**
+     * Returns the aggregated mean.
+     *
+     * @return the aggregated mean.
+     */
+    double getMean() {
+      return mean;
+    }
+
+    @Override
+    final <T> T match(
+        Function<? super MutableSum, T> p0,
+        Function<? super MutableCount, T> p1,
+        Function<? super MutableHistogram, T> p2,
+        Function<? super MutableRange, T> p3,
+        Function<? super MutableMean, T> p4,
+        Function<? super MutableStdDev, T> p5,
+        Function<? super MutableAggregation, T> defaultFunction) {
+      return p4.apply(this);
+    }
+  }
+
+  /** Calculate standard deviation on aggregated {@code MeasureValue}s. */
+  static final class MutableStdDev extends MutableAggregation {
+
+    private double sumOfSquaredDeviations = 0.0;
+    private double mean = 0.0;
+    private long count = 0;
+
+    private MutableStdDev() {
+    }
+
+    /**
+     * Construct a {@code MutableStdDev}.
+     *
+     * @return an empty {@code MutableStdDev}.
+     */
+    static MutableStdDev create() {
+      return new MutableStdDev();
+    }
+
+    @Override
+    void add(double value) {
+      count++;
+      double deltaFromMean = value - mean;
+      mean += deltaFromMean / count;
+      double deltaFromMean2 = value - mean;
+      sumOfSquaredDeviations += deltaFromMean * deltaFromMean2;
```

Comment:
Could you please add the note about the algorithm from MutableDistribution.java?